tests/backup/backup_psa_test.go

```diff
@@ -548,3 +551,597 @@ var _ = Describe("{EnableNsAndClusterLevelPSAWithBackupAndRestore}", Label(TestC
+		restrictedScheduledAppContexts []*scheduler.Context
+		privilegedScheduledAppContexts []*scheduler.Context
+		allScheduledAppContexts        []*scheduler.Context
+		namespaces                     []string
+		psaApp                         []string
+		backupNamespaceMap             = make(map[string]string)
+	)
+
+	backupLocationMap := make(map[string]string)
+	numberOfBackups, _ := strconv.Atoi(GetEnv(MaxBackupsToBeCreated, "3"))
+	params := make(map[string]string)
+	storageClassMapping := make(map[string]string)
+	ctx, err := backup.GetAdminCtxFromSecret()
+	log.FailOnError(err, "Getting admin context from secret")
+
+	JustBeforeEach(func() {
+		defaultAppList = Inst().AppList
+		log.InfoD("App list at the start of the testcase is %v", defaultAppList)
+		for _, app := range defaultAppList {
+			psaApp = append(psaApp, PSAAppMap[app])
+		}
+		Inst().AppList = psaApp
+		log.InfoD("App list for PSA %v", psaApp)
+	})
+
+	It("Take backup in restricted PSA set at a NS and restore in privilege PSA set to new NS", func() {
+		Step("Creating backup location and cloud setting", func() {
+			log.InfoD("Creating backup location and cloud setting")
+			backupLocationProviders := GetBackupProviders()
+			for _, provider := range backupLocationProviders {
+				credName = fmt.Sprintf("%s-cred-%v", provider, RandomString(10))
+				credUid = uuid.New()
+				err := CreateCloudCredential(provider, credName, credUid, BackupOrgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of cloud credential named [%s] for org [%s]  as provider %s", credName, BackupOrgID, provider))
+				backupLocationName = fmt.Sprintf("%s-backup-location-%v", provider, RandomString(10))
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = backupLocationName
+				err = CreateBackupLocation(provider, backupLocationName, backupLocationUID, credName, credUid, getGlobalBucketName(provider), BackupOrgID, "", true)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating backup location %s", backupLocationName))
+			}
+		})
+
+		Step("Registering cluster for backup", func() {
+			log.InfoD("Registering cluster for backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateApplicationClusters(BackupOrgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			clusterStatus, err = Inst().Backup.GetClusterStatus(BackupOrgID, SourceClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+			srcClusterUid, err = Inst().Backup.GetClusterUID(ctx, BackupOrgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+		})
+
+		Step("Creation of pre and post exec rules for applications", func() {
```

Comment:
Need to create rule for multiple applications , since in one of the step in single namespace multiple application is deployed